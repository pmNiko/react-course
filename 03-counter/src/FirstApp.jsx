import PropTypes from "prop-types";

const FirstApp = ({ title, subtitle = 'Subtitulo de muestra' , name = 'no name' }) => {
  return (
    <>
      <h1 data-testid='test-title' >{title}</h1>

      <p>{subtitle}</p>
      
      <p>{subtitle}</p>
      
      <p>{name}</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string
}


export default FirstApp;