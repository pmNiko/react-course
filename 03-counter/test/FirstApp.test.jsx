import { render } from '@testing-library/react';
import FirstApp from '../src/FirstApp';

describe('Pruebas en el Componente <FirstApp/>', () => { 

  // ?? Genera un snapshot en la carpeta de snaps 
  // ?? Ojo porque se vuelve demasiado estricto
  // test('Debe hacer match con el snapshopt', () => { 
  //   const title = 'Hola soy Goku!'
  //   const { container } = render(<FirstApp title={title} /> )

  //   expect(container).toMatchSnapshot();  
  // });

  test('Debe mostrar el titulo en un h1', () => { 
    const title = 'Hola soy Goku!'
    const { container, getByText ,getByTestId } = render(<FirstApp title={title} /> )

    expect( getByText(title) ).toBeTruthy();
    
    // const h1 = container.querySelector('h1');
    // expect(h1.innerHTML).toContain(title);

    expect( getByTestId('test-title') ).toBeTruthy();
    expect( getByTestId('test-title').innerHTML ).toContain(title);
  });

  // test('Debe mostrar el subtitulo enviado por props', () => { 
  //   const title = 'Hola soy Goku!';
  //   const subtitle = 'Este es un subtitulo enviado por props';
  //   const { getByText } = render(
  //     <FirstApp 
  //       title={title} 
  //       subtitle={subtitle} 
  //     /> 
  //   );

  //   expect( getByText(subtitle) ).toBeTruthy();
  // });

  test('Debe mostrar el subtitulo 2 veces', () => { 
    const title = 'Hola soy Goku!';
    const subtitle = 'Este es un subtitulo enviado por props';
    const { getAllByText } = render(
      <FirstApp 
        title={title} 
        subtitle={subtitle} 
      /> 
    );

    expect( getAllByText(subtitle) ).toHaveLength(2)
  });

});