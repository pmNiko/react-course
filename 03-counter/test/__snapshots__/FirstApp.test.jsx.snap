// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <FirstApp/> Debe hacer match con el snapshopt 1`] = `
<div>
  <h1>
    Hola soy Goku!
  </h1>
  <p>
    Subtitullo de muestra
  </p>
  <p>
    no name
  </p>
</div>
`;
